src/adaXT/predict/predict.pyx

```diff
@@ -28,16 +85,16 @@ cdef class Predict():
             X = np.expand_dims(X, axis=0)
         else:
             if X.shape[1] != self.n_features:
-                raise ValueError(f"Dimension should be {self.n_features}, got {X.shape[1]}")
+                raise ValueError(f"Dimension should be {self.n_features}, got\
```

Comment:
Do we even need a dimension check here. These functions are somewhat internal so the check should be happening in the functions the user accesses or not?

Reply:
We could impose that predict never vchecks dimensions, and then leave it to the decision tree to make sure it is done beforehand. Added an issue with this.